transcom/mymove
# B-19178 excess weight bugfix

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19178)
## [INT PR](https://github.com/transcom/mymove/pull/12816)
## [Original MAIN PR (never merged)](https://github.com/transcom/mymove/pull/12608)
## [Original INT PR](https://github.com/transcom/mymove/pull/12499)

## Summary

Excess weight was only being checked in `mto_shipment_updater.go`, when an approved shipment received an estimated weight. We added a `CheckExcessWeight` to `ApproveShipment` in `shipment_approver.go` to to ensure it is also called here when a shipment with an estimated weight is approved.

The bug existed because the front end also checks if the excess weight should be flagged, and displays the flag if so, but is unable to properly update `excess_weight_acknowledged_at` if `excess_weight_qualified_at` is null.

Please check that changes from the Original INT PR were properly reverted (ignoring changes to the CreateCustomerForm.test.jsx file as that has since been modified by a dev)

### How to test

The expected behavior is, if estimated weight is within 90% of the customer's weight allowance, the warning/flag appears on the Move Task Order page. Once the user interacts with it in any fashion (clicking x on the row, clicking on "Review Billable Weight"), they are acknowledging the risk, and should no longer be warned.
1. On a move with no approved shipments, as PRIME, enter an estimated weight below 90% of a weight allowance for a NTS/NTSR/HHG shipment
2. As a TOO, confirm:
- If the shipment was already approved, confirm that the Excess Weight warning does not appear.
- If the shipment was not already approved, approve it and confirm that the Excess Weight warning does not appear.
3. As PRIME, create a NTS/NTSR/HHG shipment. Input an estimated weight that would put the total non-PPM estimated weight of a move over 90% of the weight weight allowance
4. As a TOO confirm that the excess weight flag does NOT appear on the move task order page
5. As a TOO, approve the shipment you created in step 3
6. As a TOO, confirm the excess weight warning DOES appear, and that interacting in any way acknowledges it and it no longer appears on refresh
